src/sys/resource.rs

```diff
@@ -39,18 +40,43 @@ libc_enum!{
         #[cfg(any(target_os = "android", target_os = "linux"))]
         RLIMIT_SIGPENDING,
 
-        // Non-Linux
+        // Available on some BSD
         #[cfg(target_os = "freebsd")]
         RLIMIT_KQUEUES,
         #[cfg(target_os = "freebsd")]
         RLIMIT_NPTS,
-        #[cfg(target_os = "freebsd")]
+        #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
         RLIMIT_SBSIZE,
         #[cfg(target_os = "freebsd")]
         RLIMIT_SWAP,
     }
 }
 
+/// Get the current processes resource limits
+///
+/// A value of `None` corresponds to `RLIM_INFINITY`, which means there's no limit.
+///
+/// # Parameters
+///
+/// * `resource`: The [`Resource`] that we want to get the limits of.
+///
+/// # Examples
+///
+/// ```
+/// use nix::sys::resource::{getrlimit, Resource};
```

Comment:
In the doc test, you should put a `#` before the `use`, `fn main`, and final `}` lines so they will be hidden from the documentation.  Likewise, omit the blank line before `main`.

Reply:
Done